### PR TITLE
multiple refs for fuel-qa repo #21

### PR DESCRIPTION
**Pull request by naumvd95**

### CODE REVIEW

File: servers/partner-ci/9.x/nsxt.yaml [hunk 1]
```diff
@@ -303,11 +305,20 @@
       - test-property
     builders:
       - shell: |
+          if [[ "${{FUEL_QA_REFS}}" != "none" ]]; then
+            cd ${{WORKSPACE}}/plugin_test/fuel-qa
+            FUEL_QA_GIT_URL=$(git config --get remote.origin.url)
+            for refspec in ${{FUEL_QA_REFS}}; do
+              git fetch ${{FUEL_QA_GIT_URL}} "$refspec" && git cherry-pick FETCH_HEAD || git status
+              git log --pretty=format:"%h %ad | %s%d [%an]" --graph --date=short | head
+            done
```
Comment (rkhozinov) [inline servers/partner-ci/9.x/nsxt.yaml:314]: I think we should stop a build if needed refs aren't applied properly

File: servers/partner-ci/9.x/nsxt.yaml [hunk 2]
```diff
@@ -303,11 +305,20 @@
...
+          fi
+      - shell: |
           if [[ "${{MULTI_REFS}}" != "none" ]]; then
             cd ${{WORKSPACE}}
             for refspec in ${{MULTI_REFS}}; do
               git fetch ${{GIT_URL}} "$refspec" && git cherry-pick FETCH_HEAD || git status
-              git log --pretty=format:%h %ad | %s%d [%an] --graph --date=short | head
+              git log --pretty=format:"%h %ad | %s%d [%an]" --graph --date=short | head
```
Comment (rkhozinov) [inline servers/partner-ci/9.x/nsxt.yaml:321]: the same as above

File: servers/partner-ci/9.x/nsxv.yaml [hunk 3]
```diff
@@ -286,11 +288,20 @@
       - test-property
     builders:
       - shell: |
```
Comment (rkhozinov) [inline servers/partner-ci/9.x/nsxv.yaml:290]: let's create a marcos or yaml anchors to combine steps like that to avoid code duplication

Comment (rkhozinov) [reply servers/partner-ci/9.x/nsxv.yaml:290]: I mean builder steps those are common for some projects

File: servers/partner-ci/9.x/nsxv.yaml [hunk 4]
```diff
@@ -530,7 +541,7 @@
           #!/bin/bash
           echo "TEST_GROUP=nsxv_smoke" > property
```
Comment (rkhozinov) [inline servers/partner-ci/9.x/nsxv.yaml:542]: optimization is needed, add these options by cat

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/9.0/manila.yaml [hunk 5]
```diff
@@ -1,4 +1,32 @@
 ---
+- parameter:
+    name: plugin-specific-parameters
+    parameters:
+      - text:
+          name: DEFAULT_PARAMETERS
+          default: |
+            ENV_PREFIX=${{fuel-release}}.${{plugin-name}}
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/manila.yaml:12]: Initialize parameters directly

File: servers/partner-ci/9.0/manila.yaml [hunk 6]
```diff
@@ -176,6 +204,7 @@
           kvm_use:              '{kvm_use}'
           iso_storage:          '{iso_storage}'
           snapshots_id:         '{snapshots_id}'
+      - plugin-specific-parameters
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/manila.yaml:211]: to initialize **plugin-specific-parameters** need to pass all variables defined above in the **project** section

File: servers/partner-ci/9.0/manila.yaml [hunk 7]
```diff
@@ -242,32 +272,44 @@
             cd ${{WORKSPACE}}
             for refspec in ${{MULTI_REFS}}; do
               git fetch ${{GIT_URL}} "$refspec" && git cherry-pick FETCH_HEAD || git status
-              git log  --graph --date=short --pretty=format:%h %ad | %s%d [%an] | head
+              git log --pretty=format:"%h %ad | %s%d [%an]" --graph --date=short | head
             done
           fi
+      - shell: |
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/manila.yaml:292]: Please describe a problem the function solves

File: servers/partner-ci/9.x/nsxv.yaml (hunk 3 above)
Comment (naumvd95) [reply servers/partner-ci/9.x/nsxv.yaml:290]: seems we need to do tgis in separate request

File: servers/partner-ci/9.0/manila.yaml (hunk 7 above)
Comment (naumvd95) [reply servers/partner-ci/9.0/manila.yaml:292]: parsing by \n

File: servers/partner-ci/9.x/nsxv.yaml (hunk 4 above)
Comment (naumvd95) [reply servers/partner-ci/9.x/nsxv.yaml:542]: separate PR

File: servers/partner-ci/9.0/contrail.yaml [hunk 8]
```diff
@@ -446,11 +448,20 @@
...
+              git log --pretty=format:"%h %ad | %s%d [%an]" --graph --date=short | head
+            done
+          fi
+      - shell: |
           if [[ "${{MULTI_REFS}}" != "none" ]]; then
             cd ${{WORKSPACE}}
             for refspec in ${{MULTI_REFS}}; do
-              git fetch ${{GIT_URL}} "$refspec" && git cherry-pick FETCH_HEAD || git status
-              git log --pretty=format:'%h %ad | %s%d [%an]' --graph --date=short
+              git fetch ${{GIT_URL}} "$refspec" && git cherry-pick FETCH_HEAD ||  { git status; exit 1; }
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/contrail.yaml:463]: seems ok

File: servers/partner-ci/9.0/manila.yaml [hunk 9]
```diff
@@ -1,4 +1,36 @@
...
+    name: plugin-specific-parameters
+    parameters:
+      - string:
+          name: MULTI_REFS
+      - string:
+          name: FUEL_QA_REFS
+      - text:
+          name: DEFAULT_PARAMETERS
+          default: |
+            ENV_PREFIX=${{fuel-release}}.${{plugin-name}}
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/manila.yaml:12]: need to initialize that

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/9.x/nsxv.yaml (hunk 3 above)
Comment (rkhozinov) [reply servers/partner-ci/9.x/nsxv.yaml:290]: create a separate request

File: servers/partner-ci/9.0/manila.yaml (hunk 9 above)
Comment (naumvd95) [reply servers/partner-ci/9.0/manila.yaml:12]: separate PR

**Pull request closed without merging**